livebooks/advent_of_code_2022_day_22_tdd_elixir_binary_noggin.livemd

```diff
@@ -0,0 +1,151 @@
+  So, a path like `10R5` means "go forward 10 tiles, then turn clockwise 90 degrees, then go forward 5 tiles".
+
+You begin the path in the leftmost open tile of the top row of tiles. Initially, you are facing **to the right** (from the perspective of how the map is drawn).
+
+If a movement instruction would take you off of the map, you **wrap around** to the other side of the board. In other words, if your next tile is off of the board, you should instead look in the direction opposite of your current facing as far as you can until you find the opposite edge of the board, then reappear there.
+
+For example, if you are at `A` and facing to the right, the tile in front of you is marked `B`; if you are at `C` and facing down, the tile in front of you is marked `D`:
+
+```
+        ...#
+        .#..
+        #...
+        ....
+...#.D.....#
+........#...
+B.#....#...A
+.....C....#.
+        ...#....
+        .....#..
+        .#......
+        ......#.
+```
+
+It is possible for the next tile (after wrapping around) to be a **wall**; this still counts as there being a wall in front of you, and so movement stops before you actually wrap to the other side of the board.
+
+By drawing the **last facing you had** with an arrow on each tile you visit, the full path taken by the above example looks like this:
+
+```
+        >>v#    
+        .#v.    
+        #.v.    
+        ..v.    
+...#...v..v#    
+>>>v...>#.>>    
+..#v...#....    
+...>>>>v..#.    
+        ...#....
+        .....#..
+        .#......
+        ......#.
+```
+
+To finish providing the password to this strange input device, you need to determine numbers for your final **row, column,** and **facing** as your final position appears from the perspective of the original map. Rows start from `1` at the top and count downward; columns start from `1` at the left and count rightward. (In the above example, row 1, column 1 refers to the empty space with no tile on it in the top-left corner.) Facing is `0` for right (`>`), `1` for down (`v`), `2` for left (`<`), and `3` for up (`^`). The **final password** is the sum of 1000 times the row, 4 times the column, and the facing.
+
+In the above example, the final row is `6`, the final column is `8`, and the final facing is `0`. So, the final password is 1000 * 6 + 4 * 8 + 0: `6032`.
+
+Follow the path given in the monkeys' notes. **What is the final password?**
+
+```elixir
+defmodule DayTwentyTwoPuzzleOne do
+  defstruct []
+
+  # implement a constructor
+  def new(opts \\ %{}) do
+    struct(%__MODULE__{}, opts)
+  end
+
+  # implement reducers here
+
+  # implement converter here
+
+  def solve_one(input_string) do
+  end
+end
+```
+
+## Test
+
+```elixir
+ExUnit.start(auto_run: false)
+
+defmodule DayTwentyTwoPuzzleOneTest do
+  use ExUnit.Case, async: false
+
+  @input_string """
+
```

Comment:
Is the input supposed to be blank?

Reply:
Bah! No, it's not! FIXING!

Reply:
Corrected with an amend, force-push-with-lease